In [18]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns 

Reading the data

In [19]:
df=pd.read_csv('mymoviedb.csv',lineterminator='\n')
df

Release_Date                                 Title  \
0      2021-12-15               Spider-Man: No Way Home   
1      2022-03-01                            The Batman   
2      2022-02-25                               No Exit   
3      2021-11-24                               Encanto   
4      2021-12-22                        The King's Man   
...           ...                                   ...   
9822   1973-10-15                              Badlands   
9823   2020-10-01                      Violent Delights   
9824   2016-05-06                          The Offering   
9825   2021-03-31  The United States vs. Billie Holiday   
9826   1984-09-23                               Threads   

                                               Overview  Popularity  \
0     Peter Parker is unmasked and no longer able to...    5083.954   
1     In his second year of fighting crime, Batman u...    3827.658   
2     Stranded at a rest stop in the mountains durin...    2618.087   
3     The tale of an extraordinary family, the Madri...    2402.201   
4     As a collection of history's worst tyrants and...    1895.511   
...                                                 ...         ...   
9822  A dramatization of the Starkweather-Fugate kil...      13.357   
9823  A female vampire falls in love with a man she ...      13.356   
9824  When young and successful reporter Jamie finds...      13.355   
9825  Billie Holiday spent much of her career being ...      13.354   
9826  Documentary style account of a nuclear holocau...      13.354   

      Vote_Count  Vote_Average Original_Language  \
0           8940           8.3                en   
1           1151           8.1                en   
2            122           6.3                en   
3           5076           7.7                en   
4           1793           7.0                en   
...          ...           ...               ...   
9822         896           7.6                en   
9823           8           3.5                es   
9824          94           5.0                en   
9825         152           6.7                en   
9826         186           7.8                en   

                                   Genre  \
0     Action, Adventure, Science Fiction   
1               Crime, Mystery, Thriller   
2                               Thriller   
3     Animation, Comedy, Family, Fantasy   
4       Action, Adventure, Thriller, War   
...                                  ...   
9822                        Drama, Crime   
9823                              Horror   
9824           Mystery, Thriller, Horror   
9825               Music, Drama, History   
9826         War, Drama, Science Fiction   

                                             Poster_Url  
0     https://image.tmdb.org/t/p/original/1g0dhYtq4i...  
1     https://image.tmdb.org/t/p/original/74xTEgt7R3...  
2     https://image.tmdb.org/t/p/original/vDHsLnOWKl...  
3     https://image.tmdb.org/t/p/original/4j0PNHkMr5...  
4     https://image.tmdb.org/t/p/original/aq4Pwv5Xeu...  
...                                                 ...  
9822  https://image.tmdb.org/t/p/original/z81rBzHNgi...  
9823  https://image.tmdb.org/t/p/original/4b6HY7rud6...  
9824  https://image.tmdb.org/t/p/original/h4uMM1wOhz...  
9825  https://image.tmdb.org/t/p/original/vEzkxuE2sJ...  
9826  https://image.tmdb.org/t/p/original/lBhU4U9Eeh...  

[9827 rows x 9 columns]

Data Preprocessing 

In [20]:
df.shape

(9827, 9)

In [21]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9827 entries, 0 to 9826
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Release_Date       9827 non-null   object 
 1   Title              9827 non-null   object 
 2   Overview           9827 non-null   object 
 3   Popularity         9827 non-null   float64
 4   Vote_Count         9827 non-null   int64  
 5   Vote_Average       9827 non-null   float64
 6   Original_Language  9827 non-null   object 
 7   Genre              9827 non-null   object 
 8   Poster_Url         9827 non-null   object 
dtypes: float64(2), int64(1), object(6)
memory usage: 691.1+ KB


In [22]:
df.isna().sum()

Release_Date         0
Title                0
Overview             0
Popularity           0
Vote_Count           0
Vote_Average         0
Original_Language    0
Genre                0
Poster_Url           0
dtype: int64

In [23]:
df.duplicated().sum()

np.int64(0)

In [24]:
df.Genre.head()

0    Action, Adventure, Science Fiction
1              Crime, Mystery, Thriller
2                              Thriller
3    Animation, Comedy, Family, Fantasy
4      Action, Adventure, Thriller, War
Name: Genre, dtype: object

In [25]:
df.describe()

Popularity    Vote_Count  Vote_Average
count  9827.000000   9827.000000   9827.000000
mean     40.326088   1392.805536      6.439534
std     108.873998   2611.206907      1.129759
min      13.354000      0.000000      0.000000
25%      16.128500    146.000000      5.900000
50%      21.199000    444.000000      6.500000
75%      35.191500   1376.000000      7.100000
max    5083.954000  31077.000000     10.000000

Summary:

Given dataset contains 9827 rows and 9 columns

Out of 9 column , 3 columns (Overview, Original Language and Poster_url) will be removed because they are not useful for  analysis 

Our dataset conatains no duplicate and NaNs values 

Released_Date column has values in string datatype so we need to convert it to date format to extract only year value 

Vote_Average better be categorized for proper analysis 

Genre column has comma separated values and white spaces that needs to be handled and casted into category.

Converting and Dropping the columns

In [26]:
df["Release_Date"]=pd.to_datetime(df["Release_Date"])
df["Release_Date"].dtypes

dtype('<M8[ns]')

In [27]:
df["Release_Date"]=df["Release_Date"].dt.year
df["Release_Date"].dtypes

dtype('int32')

In [28]:
df["Release_Date"].head()

0    2021
1    2022
2    2022
3    2021
4    2021
Name: Release_Date, dtype: int32

In [29]:
cols=['Overview', 'Original_Language','Poster_Url']
df.drop(cols,axis=1,inplace=True)

In [30]:
df.head()

Release_Date                    Title  Popularity  Vote_Count  \
0          2021  Spider-Man: No Way Home    5083.954        8940   
1          2022               The Batman    3827.658        1151   
2          2022                  No Exit    2618.087         122   
3          2021                  Encanto    2402.201        5076   
4          2021           The King's Man    1895.511        1793   

   Vote_Average                               Genre  
0           8.3  Action, Adventure, Science Fiction  
1           8.1            Crime, Mystery, Thriller  
2           6.3                            Thriller  
3           7.7  Animation, Comedy, Family, Fantasy  
4           7.0    Action, Adventure, Thriller, War

In [31]:
df['Genre'].unique()

array(['Action, Adventure, Science Fiction', 'Crime, Mystery, Thriller',
       'Thriller', ..., 'Comedy, TV Movie, Romance',
       'Science Fiction, Fantasy, Family, Music',
       'War, Drama, Science Fiction'], shape=(2337,), dtype=object)

In [32]:
df['Genre']=df['Genre'].str.split(', ')
df=df.explode('Genre').reset_index(drop=True)
df.head()

Release_Date                    Title  Popularity  Vote_Count  \
0          2021  Spider-Man: No Way Home    5083.954        8940   
1          2021  Spider-Man: No Way Home    5083.954        8940   
2          2021  Spider-Man: No Way Home    5083.954        8940   
3          2022               The Batman    3827.658        1151   
4          2022               The Batman    3827.658        1151   

   Vote_Average            Genre  
0           8.3           Action  
1           8.3        Adventure  
2           8.3  Science Fiction  
3           8.1            Crime  
4           8.1          Mystery

In [33]:
#casting genre column into category
df["Genre"]=df['Genre'].astype('category')
df['Genre'].dtypes

CategoricalDtype(categories=['Action', 'Adventure', 'Animation', 'Comedy', 'Crime',
                  'Documentary', 'Drama', 'Family', 'Fantasy', 'History',
                  'Horror', 'Music', 'Mystery', 'Romance', 'Science Fiction',
                  'TV Movie', 'Thriller', 'War', 'Western'],
, ordered=False, categories_dtype=object)